### DISCUSSION TITLE
Reading SD

### DISCUSSION THREAD

**Discussion by reggievs** [cyberman54/ESP32-Paxcounter#980] [Q&A] [answered]
I'm built and uploaded the code to a Lily T Dongle S3 (with Display) and I can see the counter ticking away. Having a hard time mounting the SD (TF card) that is in the device to my Mac (M1) so I can read the CSV data. Any help would be super helpful!

**Comment by cyberman54** [comment]
How did you format the SD card?

**Reply by reggievs** [reply]
I didn't format it. It's not recognized on my M1 Mac, but I was able to get the code uploaded to it using platformio in visual studio and can confirm it's working well.

**Reply by cyberman54** [reply]
Wait, we are talking on the SD card. There is no code uploaded to it. If you flashed your paxcounter via serial cable, and enabled SD mode, and inserted an SD card, you need to format the SD card before. Otherwise it won't be identified by paxcounter, as well as by your Mac.

**Reply by cyberman54** [reply]
You need at FAT32 formatted SD card.
See https://github.com/cyberman54/ESP32-Paxcounter/blob/master/src/sdcard.cpp#L4

**Reply by reggievs** [reply]
Ah got it! It's not showing up on my Mac as a volume, I will try on PC. Once it's formatted to FAT32, I'll try again!

**Reply by reggievs** [reply]
Looks like the TF card that came with it is just outright broken, isn't mounting on my PC nor Mac. Bought a new one to test.
Other than SD, I can't transmit the data via WIFI to my computer instead from the Lily T Dongle S?

**Reply by cyberman54** [reply]
You can use the USB interface of your dongle to transmit the data, but not Wifi, because Wifi is used for sniffing.

**Reply by cyberman54** [reply]
There is no TF/SD card delivered with the Lilygo T-Dongle. What you find in the slot is just a plastic dummy. You need to replace this with a real SD card, and you need to FAT32 format this on a PC or MAC before.

**Comment by reggievs** [comment]
Got it! To use the USB interface to transmit data to my MacBook, would I
need to modify any of the configuration files?

**Reply by cyberman54** [reply]
Read the [docs](https://cyberman54.github.io/ESP32-Paxcounter/).
Create _platformio.ini_ from template _platformio_orig_s3.ini_ (copy file).
Select board type **ttgotdongledisplays3.h** in _platformio.ini_.
You will see the data on serial port.

**Reply by reggievs** [reply]
New SD card came in today. I formatted to FAT32, but CSV file after running it for 5min is empty?

**Reply by cyberman54** [reply, marked as answer]
You need to wait a while, writes to SD are asynchronous to save flash write cycles.

**Reply by reggievs** [reply]
Gotcha! So I should wait at least 5-10min after I'm done capturing data to unplug the device and then try to read the SD card?

**Reply by reggievs** [reply]
I was able to get the CSV to open after waiting 5-10min! Thank you!
